### ISSUE TITLE
Package Control: The dependency specified, bz2, is not available

### ISSUE THREAD

**Issue by darylldawn** [wbond/package_control#1114]
Hello,

Thanks for the awesome package.
I've been using Package Control ever since on Windows.
Also, I'm behind a corporate proxy.
Recently, I've started to develop on Mac (El Capitan) and get the error below while trying to load Package Control. Also, Sublime 3 shows that there are no packages available for installation.

I was actually able to use Package Control (meaning that the available for download packages are shown) prior to updating to the latest Sublime 3 Build (from 3103 to 3114).

I've tried all of the solutions based on the issues submitted here as well (reinstalling Package Control manually, deleting `Package Control.sublime-package`, etc..)

From what I understand, I think the bz2 package is not downloaded due to the error described in the debug log. 

`unzip -l ~/Library/Application\ Support/Sublime\ Text\ 3/Installed\ Packages/0_package_control_loader.sublime-package`

```
Archive:  /Users/user/Library/Application Support/Sublime Text 3/Installed Packages/0_package_control_loader.sublime-package
  Length     Date   Time    Name
 --------    ----   ----    ----
      173  05-16-16 11:04   dependency-metadata.json
     1876  05-16-16 11:04   00-package_control.py
 --------                   -------
     2049                   2 files
```

I can access packagecontrol.io via Safari in an encrypted connection, so certificates seems okay.

What is weird is that if I go to 
https://packagecontrol.io/submit?version=3.1.2&package=Package+Control&package_control_version=3.1.2&sublime_platform=osx&operation=install&sublime_version=3114
via Safari, it outputs: `{"result": "success"}` on the webpage. 

Accessing https://packagecontrol.io/channel_v3.json via Safari also shows the entire list of repositories.

This also has no problem:
`curl --proxy myproxy:myport  -o channel_v3.json https://packagecontrol.io/channel_v3.json`

```
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 2496k  100 2496k    0     0   265k      0  0:00:09  0:00:09 --:--:--  321k
```

> Package Control: Download Debug
>   URL: https://packagecontrol.io/submit?version=3.1.2&package=Package+Control&package_control_version=3.1.2&sublime_platform=osx&operation=install&sublime_version=3114
>   Timeout: 30
>   Resolved IP: [Errno 8] nodename nor servname provided, or not known
> Package Control: Urllib Debug Proxy
>   http_proxy: myproxy:myport
>   http_proxy: myproxy:myport
>   proxy_username: 
>   proxy_password: 
> Package Control: Found previously exported CA bundle at /Users/user/Library/Application Support/Sublime Text 3/Packages/User/Package Control.system-ca-bundle
> Package Control: Urllib HTTPS Debug General
>   Connecting to myproxy on port myport
> Package Control: Urllib HTTPS Debug Write
>   CONNECT packagecontrol.io:443 HTTP/1.1
>   Host: packagecontrol.io:443
>   User-Agent: Package Control v3.1.2
>   Proxy-Connection: Keep-Alive
> Package Control: Urllib HTTPS Debug Read
>   HTTP/1.0 200 Connection established
> Package Control: Urllib HTTPS Debug General
>   Upgrading connection to SSL using CA certs file at /Users/user/Library/Application Support/Sublime Text 3/Packages/User/Package Control.merged-ca-bundle
> Package Control: Error submitting usage information. HTTP exception InvalidCertificateException (Host myproxy:myport returned an invalid certificate ([SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:548))) downloading https://packagecontrol.io/submit?version=3.1.2&package=Package+Control&package_control_version=3.1.2&sublime_platform=osx&operation=install&sublime_version=3114.
> Package Control: Installing 1 missing dependency
> Package Control: Fetching list of available packages
>   Platform: osx-x64
>   Sublime Text Version: 3114
>   Package Control Version: 3.1.2
> Package Control: Download Debug
>   URL: https://packagecontrol.io/channel_v3.json
>   Timeout: 30
>   Resolved IP: [Errno 8] nodename nor servname provided, or not known
> Package Control: Urllib HTTPS Debug General
>   Connecting to myproxy on port myport
> Package Control: Urllib HTTPS Debug Write
>   CONNECT packagecontrol.io:443 HTTP/1.1
>   Host: packagecontrol.io:443
>   User-Agent: Package Control v3.1.2
>   Proxy-Connection: Keep-Alive
> Package Control: Urllib HTTPS Debug Read
>   HTTP/1.0 200 Connection established
> Package Control: Urllib HTTPS Debug General
>   Upgrading connection to SSL using CA certs file at /Users/user/Library/Application Support/Sublime Text 3/Packages/User/Package Control.merged-ca-bundle
> Package Control: Error downloading channel. HTTP exception InvalidCertificateException (Host myproxy:myport returned an invalid certificate ([SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:548))) downloading https://packagecontrol.io/channel_v3.json.
> Package Control: The dependency specified, bz2, is not available
> Package Control: Skipping automatic upgrade, last run at 2016-05-16 11:04:18, next run at 2016-05-16 12:04:18 or after

Thanks in advance!

**Comment by darylldawn** [comment]
Update: I switched back to using Sublime 2 for now. No problems with Package Control.

If there are any updates regarding Package Control with the latest build (3114) on Mac (10.11.4), 
please let me know.

Thanks!

**Comment by darylldawn** [comment]
Another update: Downgrading to Sublime 3 Build 3103 from Build 3114 has no problems with Package Control.

**Comment by wbond** [comment]
From the error message you posted, your proxy seems to be returning an invalid certificate. Or at least, the Python ssl module thinks it is invalid.

Other software on your machine may be configured differently in terms of ignoring specific certificates, or having an extra trust root.

**Comment by pgigena** [comment]
I have the same problem as the OP, and know it is being caused by my employer's proxy certificate no longer being valid (As explained by @wbond). Since I cannot work around the proxy and cannot wait for a valid certificate to be implemented, is there any way to work around this issue while still using Package Control?

**Comment by darylldawn** [comment]
Hello, thanks for the reply.

Sorry for the trouble, I am having a problem understanding. 

Package Control is installed in the same software, Sublime 3. 
It works for Build 3103, but not for Build 3114. 

Is Package Control in the different builds be accessing the proxy certificate differently?

Also, I am using a Mac. 
Package Control in Windows seems to have no problem with the upgrade.
I am not sure if this is also related to other recent Mac concerns.

Thanks again!

**Comment by wbond** [comment]
`bz2` has been included in ST3 since build 3109. I [recently updated the dependencies](9bb1c385e8fdec05570b83a36598dd678a5b2a93) to mark it as required only for older builds.

I believe your issues is something related to your network and proxy. The error message is coming from OpenSSL and indicates it can't validate the SSL certificate.

At some point in the future I'm hoping to add some debugging tools to help users figured this sort of thing out.

**Comment by deathaxe** [comment]
Various certificate issues have been fixed in 3.4.1.
